app/web_framework/cluster_base_handler.py

```diff
@@ -62,6 +66,14 @@ def _handle_request_exception(self, ex):
         self.set_status(status_code)
         self.finish()
 
+    def _get_pagination_params(self):
```

Comment:
There isn't really a good way to write tests for this method unless we have this data stored in some response. The problem is that this method gets the offset and limit from the query arguments in a request, and then does things internally with them, so there isn't really a way for us to intercept those values without explicitly having a handler that does that.

One idea is we could split this up into two methods like:

```python
def get_pagination_params(self) -> Tuple[int, int]:
    offset = self.get_query_argument('offset', self.DEFAULT_OFFSET, True)
    limit = self.get_query_argument('limit', self.DEFAULT_LIMIT, True)
    return self._validate_arguments(int(offset), int(limit))

def _validate_arguments(self, offset: int, limit: int) -> Tuple[int, int]:
    # No negative offsets or limits are accepted
    non_negative_offset = max(offset, 0)
    non_negative_limit = max(offset, 0)
    return non_negative_limit, min(non_negative_limit, self.MAX_LIMIT)
```

Then we can easily unit test `_validate_arguments`. I think this would be a good idea for testing and also wouldn't hurt to separate the methods anyways -- any opinions before I do this on whether or not we should?